predicates/traffic/segment_test.go

```diff
@@ -51,20 +51,30 @@ func requestWithR(r float64) *http.Request {
 	return req
 }
 
-func doN(t *testing.T, n int, client *proxytest.TestClient, request func() *http.Request) map[int]int {
-	codes := make(map[int]int)
+// doN performs a number of requests and returns the number of responses for each status code and
+// a total number of requests performed.
+// Results use float64 type to simplify fractional comparisons.
+func doN(t *testing.T, client *proxytest.TestClient, request func() *http.Request) (map[int]float64, float64) {
```

Comment:
Where is the N?
Maybe better to call it `do` ?

Reply:
it is to highlight that it does multiple requests